src/neural/loader.cc

```diff
@@ -104,6 +104,75 @@ std::string DecompressGzip(const std::string& filename) {
   return buffer;
 }
 
+void MovePolicyHead(WeightsFile* file) {
+  bool attn_policy = file->format().network_format().policy() ==
+    pblczero::NetworkFormat::POLICY_ATTENTION;
+  auto vanilla = file->mutable_weights()->mutable_policy_heads()->mutable_vanilla();
+  auto mutable_weights = file->mutable_weights();
+  if (attn_policy && file->weights().has_ip_pol_b()) {
+    // For attention policy weights, ip_pol_w and ip_pol_b (embedding weights)
+    // are moved to the main "policy_heads" struct, where all policy heads
+    // share them.
+    auto heads = file->mutable_weights()->mutable_policy_heads();
+    *heads->mutable_ip_pol_w() = file->weights().ip_pol_w();
+    *heads->mutable_ip_pol_b() = file->weights().ip_pol_b();
+    mutable_weights->mutable_ip_pol_w()->Clear();
+    mutable_weights->mutable_ip_pol_b()->Clear();
+
+    // Some older attention policy nets have policy encoders.
+    for (auto enc : file->weights().pol_encoder()) {
+      *vanilla->add_pol_encoder() = enc;
+    }
+    vanilla->set_pol_headcount(file->weights().pol_headcount());
+  }
+
+  // Macro to move remaining shared weights around.
```

Comment:
Would this maybe work? (Not sure whether it's less hacky, but at least it doesn't have macros)

```cpp
  vanilla->MergeFromString(file-weights().OutputAsString());
```

UPD.. Just reallized it's copying from different proto.
Then let's keep as it is, I cannot think of anything nicer.